src/master-playlist-controller.js

```diff
@@ -846,39 +866,48 @@ export class MasterPlaylistController extends videojs.EventTarget {
    * @return {TimeRange} the seekable range
    */
   seekable() {
+    if (!this.seekable_) {
+      return videojs.createTimeRanges();
+    }
+
+    return this.seekable_;
+  }
+
+  onSyncInfoUpdate_() {
     let media;
     let mainSeekable;
     let audioSeekable;
 
     if (!this.masterPlaylistLoader_) {
-      return videojs.createTimeRanges();
+      return;
     }
+
     media = this.masterPlaylistLoader_.media();
+
     if (!media) {
-      return videojs.createTimeRanges();
+      return;
     }
 
-    mainSeekable = Hls.Playlist.seekable(media,
-                                         this.masterPlaylistLoader_.expired_);
+    mainSeekable = Hls.Playlist.seekable(media);
     if (mainSeekable.length === 0) {
-      return mainSeekable;
+      return;
     }
 
     if (this.audioPlaylistLoader_) {
-      audioSeekable = Hls.Playlist.seekable(this.audioPlaylistLoader_.media(),
-                                            this.audioPlaylistLoader_.expired_);
+      audioSeekable = Hls.Playlist.seekable(this.audioPlaylistLoader_.media());
       if (audioSeekable.length === 0) {
-        return audioSeekable;
+        return;
```

Comment:
I think you're right (on this and the comment above), we do want the intersection if an audio playlist is available.